core/templates/pages/exploration-editor-page/exploration-editor-page.component.ts

```diff
@@ -346,11 +348,51 @@ export class ExplorationEditorPageComponent implements OnInit, OnDestroy {
         }
       }
 
-      // Initialize changeList by draft changes if they exist.
+      // Initialize changeList by draft changes if they exist,
+      // and initialize the entity translations by draft changes
+      // if they exist.
       if (explorationData.draft_changes !== null) {
         this.changeListService.loadAutosavedChangeList(
           explorationData.draft_changes
         );
+
+        for (let i in explorationData.draft_changes) {
+          let changeDict = explorationData.draft_changes[i];
+
+          if (changeDict.cmd === 'edit_translation') {
+            // Create the entity translation objects first if they don't exist.
+            if (
+              !this.entityTranslationsService.languageCodeToEntityTranslations.hasOwnProperty(
+                changeDict.language_code
+              )
+            ) {
+              this.entityTranslationsService.languageCodeToEntityTranslations[
+                changeDict.language_code
+              ] = EntityTranslation.createFromBackendDict({
+                entity_id: this.explorationId,
+                entity_type: 'exploration',
+                entity_version: explorationData.version,
+                language_code: changeDict.language_code,
+                translations: {},
+              });
+            }
+
+            // Update the translations appropriately, via latest draft changes.
+            this.entityTranslationsService.languageCodeToEntityTranslations[
+              changeDict.language_code
+            ].updateTranslation(
+              changeDict.content_id,
+              TranslatedContent.createFromBackendDict(changeDict.translation)
+            );
+          } else if (changeDict.cmd === 'remove_translations') {
+            for (let language in this.entityTranslationsService
+              .languageCodeToEntityTranslations) {
+              this.entityTranslationsService.languageCodeToEntityTranslations[
+                language
+              ].removeTranslation(changeDict.content_id);
```

Comment:
Would [forEach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) work here?

Reply:
Likely, do we prefer those more than normal loops? I don't see a problem with this way since its pretty straightforward

Reply:
Per discussion with @Vir-8, given that this is an object, it may involve using Object.keys(..) and then performing forEach over that array. This might be more complicated than a regular for loop, so we've decided to leave this as-is.

Reply:
Just a thought, we could have a utility function within entityTranslationsService that removes translations in all languages for a given content ID -- avoids some code duplication.

Reply:
Done - turned it into a utility function